infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -517,50 +513,65 @@ public void updateSyncStatus(String clusterId, ReplicationStatusVal.SyncType las
             CorfuStoreEntry<ReplicationStatusKey, ReplicationStatusVal, Message> record = txn.getRecord(replicationStatusTable, key);
             if (record.getPayload() != null) {
-                ReplicationStatusVal previous = record.getPayload();
-                syncInfo = previous.getSnapshotSyncInfo();
+                previous = record.getPayload();
+                snapshotStatus = previous.getSnapshotSyncInfo();
             }
             txn.commit();
         }
 
-        if (type == ReplicationStatusVal.SyncType.LOG_ENTRY) {
-            if (syncInfo == null){
-                log.warn("setReplicationStatusTable during LOG_ENTRY sync, " +
-                        "previous status is not present for cluster: {}", clusterId);
-                syncInfo = SnapshotSyncInfo.newBuilder().build();
+        if (type == SyncType.LOG_ENTRY) {
+            if (previous != null &&
+                    (previous.getStatus().equals(SyncStatus.NOT_STARTED)
+                            || snapshotStatus.getStatus().equals(SyncStatus.STOPPED))) {
+                log.info("syncStatusPoller :: skip replication status update, log entry replication is {}", previous.getStatus());
+                // Skip update of sync status, it will be updated once replication is resumed or started
+                return;
+            }
+
+            if (snapshotStatus == null){
+                log.warn("syncStatusPoller [logEntry]:: previous snapshot status is not present for cluster: {}", clusterId);
+                snapshotStatus = SnapshotSyncInfo.newBuilder().build();
             }
 
             current = ReplicationStatusVal.newBuilder()
                     .setRemainingEntriesToSend(remainingEntries)
                     .setSyncType(type)
                     .setStatus(SyncStatus.ONGOING)
-                    .setSnapshotSyncInfo(syncInfo)
+                    .setSnapshotSyncInfo(snapshotStatus)
                     .build();
 
             try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
                 txn.putRecord(replicationStatusTable, key, current, null);
                 txn.commit();
             }
             
-            log.debug("setReplicationStatusTable: clusterId: {}, remainingEntries: {}, type: {}, syncInfo: {}",
-                    clusterId, remainingEntries, type, syncInfo);
-        } else if (type == ReplicationStatusVal.SyncType.SNAPSHOT) {
-            SnapshotSyncInfo currentSyncInfo;
-            if (syncInfo == null){
-                log.warn("setReplicationStatusTable during SNAPSHOT sync, " +
-                        "previous status is not present for cluster: {}", clusterId);
-                currentSyncInfo = SnapshotSyncInfo.newBuilder().build();
+            log.debug("syncStatusPoller :: Log Entry status set to ONGOING, clusterId: {}, remainingEntries: {}, " +
+                            "snapshotSyncInfo: {}", clusterId, remainingEntries, snapshotStatus);
+        } else if (type == SyncType.SNAPSHOT) {
```

Comment:
does this 'else if' correspond to the if block on L540?  Shouldnt it correspond to the outer if block starting at L531?

Reply:
It is, L540 closes at L543